File: python/src/endpoints/db_delete.py
import os
from worqhat import Worqhat


def delete_inactive_users() -> None:
    """Delete inactive users."""
    # Initialize the client with your API key
    client = Worqhat(
        api_key=os.environ.get("WORQHAT_API_KEY"),  # Using environment variables for security
        environment=os.environ.get("WORQHAT_ENVIRONMENT", "production"),  # Defaults to production
    )

    try:
        response = client.db.delete_records(
            table="users",  # The table to delete from
            where={
                # The condition to match records
                "status": "inactive",
            },
        )

        # Handle the successful response
        print(f"Deleted {response.deleted_count} inactive users")
        print(f"Message: {response.message}")
    except Exception as e:
        # Handle any errors
        print(f"Error deleting users: {str(e)}")


def delete_old_completed_tasks() -> None:
    """Delete old completed tasks with multiple conditions."""
    # Initialize the client with your API key
    client = Worqhat(
        api_key=os.environ.get("WORQHAT_API_KEY"),
        environment=os.environ.get("WORQHAT_ENVIRONMENT", "production"),  # Defaults to production
    )

    try:
        # Get date from 30 days ago
        from datetime import datetime, timedelta
        thirty_days_ago = (datetime.now() - timedelta(days=30)).isoformat()

        response = client.db.delete_records(
            table="tasks",  # The table to delete from
            where={
                # Multiple conditions to match records
                "status": "completed",
                "priority": "low",
            },
        )

        # Handle the successful response
        print(f"Deleted {response.deleted_count} old completed tasks")
        print(f"Message: {response.message}")
    except Exception as e:
        print(f"Error deleting tasks: {str(e)}")


def db_delete() -> None:
    """Run all delete examples."""
    delete_inactive_users()
    delete_old_completed_tasks()

File: node-typescript/src/endpoints/dbDelete.ts
import Worqhat from "worqhat";

export async function deleteTaskById(taskId: number) {
  // Initialize the client with your API key - matching smoke test
  const apiKey = process.env.WORQHAT_API_KEY;
  if (!apiKey) {
    throw new Error("WORQHAT_API_KEY environment variable is required");
  }

  const client = new Worqhat({
    apiKey, // Always use environment variables for API keys
    environment: process.env.WORQHAT_ENVIRONMENT || "production", // Defaults to production
  });

  try {
    // Call the deleteRecords method - matching smoke test
    const response = await client.db.deleteRecords({
      table: "tasks", // The table to delete from - matching smoke test
      where: {
        // The condition to match records - matching smoke test
        id: taskId,
      },
    });

    // Handle the successful response
    console.log(`Deleted task with ID: ${taskId}`);
    console.log(`Message: ${response.message}`);
    return response;
  } catch (error) {
    // Handle any errors
    const errorMessage = error instanceof Error ? error.message : String(error);
    console.error("Error deleting task:", errorMessage);
    throw error;
  }
}

// Export a function to run all examples (for backward compatibility)
export async function dbDelete(taskId: number) {
  // Call the function - matching smoke test
  await deleteTaskById(taskId);
}
